src/modules/fancyzones/lib/ZoneWindow.cpp

```diff
@@ -1,765 +1,774 @@
+
+    IFACEMETHODIMP MoveSizeEnter(HWND window, bool dragEnabled) noexcept;
+    IFACEMETHODIMP MoveSizeUpdate(POINT const& ptScreen, bool dragEnabled) noexcept;
+    IFACEMETHODIMP MoveSizeEnd(HWND window, POINT const& ptScreen) noexcept;
+    IFACEMETHODIMP MoveSizeCancel() noexcept;
+    IFACEMETHODIMP_(bool) IsDragEnabled() noexcept { return m_dragEnabled; }
+    IFACEMETHODIMP_(void) MoveWindowIntoZoneByIndex(HWND window, int index) noexcept;
+    IFACEMETHODIMP_(void) MoveWindowIntoZoneByDirection(HWND window, DWORD vkCode) noexcept;
+    IFACEMETHODIMP_(void) CycleActiveZoneSet(DWORD vkCode) noexcept;
+    IFACEMETHODIMP_(std::wstring) UniqueId() noexcept { return { m_uniqueId }; }
+    IFACEMETHODIMP_(std::wstring) WorkAreaKey() noexcept { return { m_workArea }; }
+    IFACEMETHODIMP_(void) SaveWindowProcessToZoneIndex(HWND window) noexcept;
+    IFACEMETHODIMP_(IZoneSet*) ActiveZoneSet() noexcept { return m_activeZoneSet.get(); }
+
+protected:
+    static LRESULT CALLBACK s_WndProc(HWND window, UINT message, WPARAM wparam, LPARAM lparam) noexcept;
+
+private:
+    void ShowZoneWindow() noexcept;
+    void HideZoneWindow() noexcept;
+    void LoadSettings() noexcept;
+    void InitializeZoneSets(MONITORINFO const& mi) noexcept;
+    void CalculateZoneSet() noexcept;
+    void UpdateActiveZoneSet(_In_opt_ IZoneSet* zoneSet) noexcept;
+    LRESULT WndProc(UINT message, WPARAM wparam, LPARAM lparam) noexcept;
+    void OnPaint(wil::unique_hdc& hdc) noexcept;
+    void OnKeyUp(WPARAM wparam) noexcept;
+    winrt::com_ptr<IZone> ZoneFromPoint(POINT pt) noexcept;
+    void ChooseDefaultActiveZoneSet() noexcept;
+    void CycleActiveZoneSetInternal(DWORD wparam, Trace::ZoneWindow::InputMode mode) noexcept;
+    void FlashZones() noexcept;
+
+    winrt::com_ptr<IZoneWindowHost> m_host;
+    HMONITOR m_monitor{};
+    std::wstring m_uniqueId; // Parsed deviceId + resolution + virtualDesktopId
+    wchar_t m_workArea[256]{};
+    wil::unique_hwnd m_window{};
+    HWND m_windowMoveSize{};
+    bool m_drawHints{};
+    bool m_flashMode{};
+    bool m_dragEnabled{};
+    winrt::com_ptr<IZoneSet> m_activeZoneSet;
+    std::vector<winrt::com_ptr<IZoneSet>> m_zoneSets;
+    winrt::com_ptr<IZone> m_highlightZone;
+    WPARAM m_keyLast{};
+    size_t m_keyCycle{};
+    static const UINT m_showAnimationDuration = 200; // ms
+    static const UINT m_flashDuration = 700; // ms
+};
+
+ZoneWindow::ZoneWindow(HINSTANCE hinstance)
+{
+    WNDCLASSEXW wcex{};
+    wcex.cbSize = sizeof(WNDCLASSEX);
+    wcex.lpfnWndProc = s_WndProc;
+    wcex.hInstance = hinstance;
+    wcex.lpszClassName = L"SuperFancyZones_ZoneWindow";
+    wcex.hCursor = LoadCursorW(nullptr, IDC_ARROW);
+    RegisterClassExW(&wcex);
+}
+
+bool ZoneWindow::Init(IZoneWindowHost* host, HINSTANCE hinstance, HMONITOR monitor, std::wstring uniqueId, bool flashZones)
+{
+    if (!host)
+    {
+        return false;

```

Comment:
Why is it possible for `host` to be null now?

Reply:
You're right, I don't see how it can be possible, we are calling this from `FancyZones` and passing `this` for host value. We can remove this check.

Reply:
Is it guaranteed that this scenario will never be changed and `nullptr` will never be passed here? If so, then there's no need to check.

Reply:
Those nullptr checks seem innocent enough, however, I'm concerned about them, because they'll most likely confuse future readers and entice a certain coding style. I'm not going to guarantee a certain future for `IZoneWindowHost`, of course, but I can describe a high probability future scenario as a result of adding many "always true" nullptr checks I've seen multiple times:

We **know** for certain now that those checks are not needed except for new unit tests, but will we remember that and be certain in 3/6/... months? 
- Someone unfamiliar with the code will have to track all calls of this method only to be confused with its purpose.
- They'll write nullptr checks for any new code just to be safe. 
- Gradually, we'll have those nullptr checks everywhere for each pointer argument, making them basically noise without meaning. 

We'll lose valuable lifetime object information and function invariants by that point - the code will be harder to read and maintain. It would be also hard to remove those nullptr checks during future refactoring, because the PR author will get difficult and time-consuming questions like "are we sure it's safe to remove them?" etc.

Reply:
Can we refactor the tests so we can remove the checks and keep the tests?

Reply:
What would be expectation from test that is passing `nullptr` for host in this particular scenario? Exception thrown?

Reply:
Let's delete them? As it's impossible to happen ?

Reply:
Of course, for the cases where it is impossible...

Reply:
If the test is just testing the pointer, yes sure let's remove the test. If the test is testing something else, it can always mock the object.